rockchip: add support for Radxa Rock Pi 4

mj22226 opened the pull request:
This PR add support for Radxa Rock Pi 4
The only part not yet working is the wifi chip BCM43456c5-AP6256.
[Rock Pi 4 boot log](https://gist.github.com/mj22226/f12347dd4467d7a7a13867b87f7478c6)

adschm commented on target/linux/rockchip/image/armv8.mk:
```diff
@@ -15,6 +15,15 @@ define Device/friendlyarm_nanopi-r2s
 endef
 TARGET_DEVICES += friendlyarm_nanopi-r2s
 
+define Device/radxa_rock-pi-4
```
Please move down according to alphabetic sorting.

adschm commented on target/linux/rockchip/image/armv8.mk:
```diff
@@ -15,6 +15,15 @@ define Device/friendlyarm_nanopi-r2s
 
+define Device/radxa_rock-pi-4
+  DEVICE_VENDOR := Radxa
+  DEVICE_MODEL := Rock Pi 4
```
You talk about "Model B v1.3" in the commit message. Is that irrelevant?

mj22226 replied:
I just wanted to make it clear on the model version I'm testing.

mj22226 committed:
rockchip: add support for Radxa Rock Pi 4

This adds basic support for Radxa Rock Pi 4

Specification:
  - RAM:      1 GB/ 2 GB/4 GB LPDDR4

  - SoC:      Rockchip RK3399
  - CPU:      64bit hexa core processor
              Dual Cortex-A72, freqency 1.8Ghz
              with quad Cortex-A53, frequency 1.4Ghz
  - USB:      USB 3.0 OTG x1
              hardware switch for host/device switch, upper one
              USB 3.0 HOST x1
              dedicated USB3.0 channel, lower one
              USB 2.0 HOST x2
  - Ethernet: 1x GbE
  - Storage:  eMMC module
              uSD card
              M.2 SSD
  - Wireless: 802.11 ac wifi
              Bluetooth 5.0
              currently not supported

 firmware Installation
 ======================
 gzip -d xxx.img.gz, then dd the .img to SD/eMMC
 ======================

 Device Tested: ROCK PI 4 Model B v1.3

Signed-off-by: Marty Jones <mj8263788@gmail.com>

blocktrron commented on target/linux/rockchip/image/armv8.mk:
```diff
@@ -23,3 +23,12 @@ define Device/pine64_rockpro64
+  DEVICE_MODEL := Rock Pi 4
+  SOC := rk3399
+  UBOOT_DEVICE_NAME := rock-pi-4-rk3399
+  IMAGE/sysupgrade.img.gz := boot-common | boot-script | pine64-img | gzip | append-metadata
```
Have you tried adding `brcmfmac-firmware-4329-sdio` and `kmod-brcmfmac` as packages to enable the WiFI?

mj22226 replied:
brcmfmac-firmware-4329-sdio does't work, I have the WiFi working with brcmfmac-firmware-43456-sdio from armbian, can I add this to the PR https://github.com/mj22226/openwrt/commit/95564b855f3daa8ac5769c4bc7028d8fc1537999

blocktrron reviewed (changes requested):
The platform commit has an uppercase `A` after the platform prefix. the first word after the prefix should be lower case.

mj22226 committed:
uboot-rockchip: add Rock Pi 4 support

Add support for the  Raxda Rock Pi 4.

Signed-off-by: Marty Jones <mj8263788@gmail.com>
